agent/php_txn.c

```diff
@@ -669,6 +669,58 @@ static void nr_php_txn_send_metrics_once(nrtxn_t* txn TSRMLS_DC) {
 #undef FMT_BOOL
 }
 
+void nr_php_txn_create_agent_version_metric(nrtxn_t* txn, const char* version) {
+  char* metric_name = NULL;
+
+  if (NULL == txn) {
+    return;
+  }
+
+  if (nr_strempty(version)) {
+    return;
+  }
+
+  metric_name = nr_formatf("Supportability/PHP/AgentVersion/%s", version);
+  nrm_force_add(NRTXN(unscoped_metrics), metric_name, 0);
+  nr_free(metric_name);
+}
+
+void nr_php_txn_create_php_version_metric(nrtxn_t* txn, const char* version) {
+  char* metric_name = NULL;
+
+  if (NULL == txn) {
+    return;
+  }
+
+  if (nr_strempty(version)) {
+    return;
+  }
+
+  metric_name = nr_formatf("Supportability/PHP/Version/%s", version);
```

Comment:
php version and agent version are things that will be known at minit time.
can't we just compute it once at start up, make it a global variable, then create a metric as needed instead of recomputing a known value on every txn?

Reply:
Based on a suggestion by zsistla the agent version metric name is now computed at compile time - added in commit [db1ee0dc8](https://github.com/newrelic/newrelic-php-agent/pull/947/commits/db1ee0dc88fb8f658d3dec7f08bcc4b8793c81f0).
My thought on the php version metric name creation is that these are very fast operations (profiling shows 10's of thousands per msec) so the addition of this one (which happens only once per request at the end of the request, after PHP has serviced the request) is not enough to merit changing its creation.